dbms/src/Storages/Page/V3/LogFile/LogWriter.cpp

```diff
@@ -56,8 +58,9 @@ void LogWriter::resetBuffer()
 
 LogWriter::~LogWriter()
 {
-    log_file->fsync();
-    log_file->close();
+    // In case of forgetting flush when `manual_flush` is true, should do nothing when `LogWriter` is used correctly
+    flush(/* write_limiter */ nullptr, /* background */ false);
```

Comment:
I think flush is too heavy for a deconstructor function. It is much likely throwing an I/O error than fsync in the deconstructor function. I think leaving it to the user with `manual_flush` calling `flush` explicitly is OK.

Reply:
For example, in WALStore::saveSnapshot:
https://github.com/pingcap/tiflash/blob/c85332e14abcd34953837a45bab7ba48898e4544/dbms/src/Storages/Page/V3/WALStore.cpp#L201-L206

If we do flush in `~LogWriter` and `compact_log->flush` throw an I/O exception, then when exiting `WALStore::saveSnapshot`, `~LogWriter` is called and another I/O exception is thrown. An I/O exception thrown in the deconstructor will make the process exit.

Reply:
Fixed.